### ISSUE TITLE
State on and state off inverted

### ISSUE THREAD

**Issue by pilot1981** [ollo69/ha-samsungtv-smart#82]
HASSIO 2021.5.3 on raspberry 3+

Tv Samsung frame 2020

When I send power on command: it go in art mode

when I send power off command: it go in ON state.

regards

**Comment by pilot1981** [comment]
why I don't see art mode state in samsung smartthings hub web portal?


![image](https://user-images.githubusercontent.com/46350465/121039443-5f7b7f80-c7b1-11eb-9678-4c31f57a600c.png)

**Comment by ollo69** [comment]
You should activate debug log for integration and attach here log file, so that I can analyze what happen.

**Comment by pilot1981** [comment]
[log_samsung.txt](https://github.com/ollo69/ha-samsungtv-smart/files/6610420/log_samsung.txt)

**Comment by ollo69** [comment]
What enviroment / hardware are u using? What version of python? I see in your log OSError when sending the command, that is very strange.

**Comment by pilot1981** [comment]
raspberry pi 3+ with HASSIO core-2021.6.2 python 3.8.10

what is OSE?

If I use it:

`  - platform: template
    switches:
      tv_art_mode:
        friendly_name: "TV Art Mode"
        value_template: "{{ state_attr('media_player.samsung_the_frame_65_tizen', 'art_mode_status') == 'on' }}"
        turn_on:
          - service: media_player.turn_on
            entity_id: media_player.samsung_the_frame_65_tizen
          - wait_template: "{{ (is_state('media_player.samsung_the_frame_65_tizen', 'on') and state_attr('media_player.samsung_the_frame_65_tizen', 'art_mode_status') == 'off') or (state_attr('media_player.samsung_the_frame_65_tizen', 'media_content_type') == 'video' or state_attr('media_player.samsung_the_frame_65_tizen', 'app_id') == 'TV/HDMI') }}"
            timeout: '00:02:00'
            continue_on_timeout: false
          - delay: '00:00:01'
          - service: media_player.play_media
            data_template:
              entity_id: media_player.samsung_the_frame_65_tizen
              media_content_id: 'KEY_POWER'
              media_content_type: 'send_key'
        turn_off:
          - service: media_player.turn_off
            entity_id: media_player.samsung_the_frame_65_tizen`

when I go to switch on tv_art_mode sometime TV power on with lastest TV channel....

**Comment by ollo69** [comment]
`OSError` is a python exception.
Why you are using a so complex template? Use the services:

- media_player.turn_on
- media_player.turn_off
- samsungtv_smart.set_art_mode

**Comment by pilot1981** [comment]
I’m not able to use these services in  HA switches…

**Comment by pilot1981** [comment]
ok, I will try with this one:

  - platform: template
    switches:
      samsung_tv_power:
        value_template: "{{ is_state('media_player.samsung_tv', 'on') }}"
        turn_on:
          service: media_player.turn_on
          target:
            entity_id: media_player.samsung_tv
        turn_off:
          service: media_player.set_art_mode
          target:
            entity_id: media_player.samsung_tv

**Comment by pilot1981** [comment]
for the OSE error what I have to do?

**Comment by ollo69** [comment]
```
switch:
  - platform: template
    switches:
        samsung_tv_power:
            value_template: "{{ is_state('media_player.samsung_tv', 'on') }}"
            turn_on:
                service: media_player.turn_on
                target:
                    entity_id: media_player.samsung_tv
            turn_off:
                service: samsungtv_smart.set_art_mode
                target:
                    entity_id: media_player.samsung_tv
```

					


> for the OSE error what I have to do?

Nothing for now, test new template

**Comment by pilot1981** [comment]
It worked fine for a couple of day…but now calling services on and off states are already  inverted….

**Comment by pilot1981** [comment]
Hi @ollo69,
I discovered a thing:
1) when problem of inverted states
2) if I force power outage on TV, then HA comand/state come back to normal behaviour…

this can help your work? Can you fix it?

you need some other tests?

**Comment by ollo69** [comment]
I need debug log to investigate this issue and understand what happen in HA <-> TV communication.
Enable debug log for the integration and reproduce the problem, than attach the log to this thread.

**Comment by pilot1981** [comment]
problem seems to happen random, I try to replicate issue and then send you logs...

**Comment by pilot1981** [comment]
[log samsung tv frame ha.txt](https://github.com/ollo69/ha-samsungtv-smart/files/6672808/log.samsung.tv.frame.ha.txt)

@ollo69 logs in attachments

**Comment by pilot1981** [comment]
First switch on seems to be wrong , then all worked fine until last switch off..

**Comment by pilot1981** [comment]
Example:

When screen is black (tv powered off / art mode off), on HA I see tv powered on!

**Comment by pilot1981** [comment]
@ollo69 any ideas?

**Comment by cappee** [comment]
Hi, I don’t have a TV with art mode but I also have the same problem 😭
I tried to flip state but after some seconds that tv is on and also on HA is on it flips again 😪😪

**Comment by simonep2001** [comment]
> Hi, I don’t have a TV with art mode but I also have the same problem 😭
> I tried to flip state but after some seconds that tv is on and also on HA is on it flips again 😪😪

I have the same problem, i turn on the tv with the button and after some seconds it returns "off" but the tv is on
@ollo69

**Comment by pilot1981** [comment]
hi @ollo69 , any news for me?

**Comment by ollo69** [comment]
Sorry but I'm really busy on this period. I will analyze this as soon I will have some additional free time.

**Comment by ollo69** [comment]
Closing this issue, problem should be fixed as reported in issue #82
